browser/scripts/ui-core.js

```diff
@@ -436,6 +462,120 @@ VizorUI.prototype.onKeyPress = function(e) {
 			break;
 	}
 
+	// keys for program-mode (noodles visible) only:
+	if (!E2.app.worldEditor.isActive()) {
+		switch(key) {
+			case uiKeys.gotoParentGraph:
+				if (E2.core.active_graph.parent_graph)
+					E2.app.onGraphSelected(E2.core.active_graph.parent_graph);
+				break;
+			case uiKeys.viewSource:
+			case 'shift+' + uiKeys.viewSource:
+			case 'alt+shift+' + uiKeys.viewSource:	// .fi
+				this.viewSource();
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesUp:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					0, -10);
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesDown:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					0, 10);
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesLeft:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					-10, 0);
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesRight:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					10, 0);
+				e.preventDefault();
+				break;
+		}
+
+		if (key >= uiKeys.number0 && key <= uiKeys.number9) { // 0-9
```

Comment:
elegant, I'd prefer keeping consistent with the boring pattern above, and to make the keypad keys work (e.g. don't check by keyCode). could we squeeze this in this PR? please?